.dotnet/src/Generated/Internal/ClientPipelineExtensions.cs

```diff
@@ -26,7 +29,15 @@ public static async ValueTask<PipelineResponse> ProcessMessageAsync(this ClientP
                 return message.Response;
             }
 
-            throw new ClientResultException(message.Response);
+            if (_customAsyncExceptionMessageCreator is not null)
```

Comment:
I see that, even with the added partial customization layer, we've made modifications to "generated code".  I'm totally cool with that for this library, but I'd like to make sure that this isn't a proposal that this is what the generator should output going forward.  @trrwilson, can you confirm?

Reply:
I think we need *some* change in the generated `ClientPipelineExtensions` class (or the ability to replace it with custom code), as I'm not sure how else we'd intercept the exception instantiation behavior. If we can just define our own `ClientPipelineExtensions` class and codegen will automatically suppress the emitted/imported portion when present (as is the case for most patterns), then I think we 100% won't need changes in the generated/template code, as we'd just wholesale replace `ProcessMessage` and `ProcessMessageAsync`. Otherwise, is there another way we'd be able to insert logic between the pipeline send and exception instantiation?

Reply:
Would we need it if we solved https://github.com/Azure/azure-sdk-for-net/issues/42823?

Reply:
Revisiting this at last.

- In light of our codegen progress, I agree that modifying the `/Generated/Internal/` file isn't going to work.
- The linked issue to emplace the buffered response content into the exception message would significantly improve things, *but*...
- ...emplaced, raw content would still not be ideal, given we'd like to deserialize when a schema is available (as it is in this case)